### PR TITLE
Introduce FEATURE_GDBJIT_SYMTAB

### PR DESCRIPTION
**Pull request by kbaladurin**
This commit adds new feature FEATURE_GDBJIT_SYMTAB. When it's set ELF file with symtab section will be generated for all JITed methods.

It helps lldb and gdb correctly unwind frames of JITed methods as they can use assembly language inspection to determine location of saved registers on stack.

### CODE REVIEW

**Comment by kbaladurin** [comment]
cc: @ayuckhulk @Dmitri-Botcharnikov @chunseoklee

**Comment by noahfalk** [comment]
Fine by me. I haven't been scrutinizing any of the gdbjit stuff too closely fyi. I assume if the folks using it are happy and it is contained behind its ifdef then we are good.

File: src/vm/gdbjit.cpp [hunk 1]
```diff
@@ -1730,22 +1730,24 @@ struct Elf_Symbol {
     Elf_Symbol() : m_name(nullptr), m_off(0), m_value(0), m_section(0), m_size(0) {}
 };
 
-static int countFuncs(const SymbolsInfo *lines, int nlines)
+template <class T, class U>
+static int countFuncs(NewArrayHolder<T> &arr, int n, U T::*ilOffset)
```
Comment (janvorli) [inline src/vm/gdbjit.cpp]: This is unnecessarily complex. You don't need to pass in the ilOffset member pointer. The following works just fine (provided you also change the callers to not to pass the iloffset argument). Could you please change it that way?
```C++
template <class T>
static int countFuncs(T &arr, int n)
{
    int count = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i].ilOffset == ICorDebugInfo::PROLOG)
        {
            count++;
        }
    }
    return count;
}

template <class T>
static int getNextPrologueIndex(int from, T &arr, int n)
{
    for (int i = from; i < n; ++i) {
        if (arr[i].ilOffset == ICorDebugInfo::PROLOG)
        {
            return i;
        }
    }
    return -1;
}

```

Comment (kbaladurin) [reply src/vm/gdbjit.cpp]: Thank you. I've removed unnecessary parameter from this functions.

**Pull request merged**